scalafix-core/shared/src/main/scala/scalafix/rule/Rule.scala

```diff
@@ -69,24 +69,24 @@ abstract class Rule(ruleName: RuleName) { self =>
   final def andThen(other: Rule): Rule = merge(other)
 
   /** Returns string output of applying this single patch. */
-  final def apply(ctx: RuleCtx): String = apply(ctx, fixWithName(ctx))
+  final def apply(ctx: RuleCtx): String =
+    apply(ctx, fixWithName(ctx))
   final def apply(
       input: Input,
       config: ScalafixConfig = ScalafixConfig.default): String = {
     val ctx = RuleCtx(config.dialect(input).parse[Source].get, config)
     apply(ctx, fixWithName(ctx))
   }
   final def apply(input: String): String = apply(Input.String(input))
-  final def apply(ctx: RuleCtx, patch: Patch): String = {
+  final def apply(ctx: RuleCtx, patch: Patch): String =
     apply(ctx, Map(name -> patch))
-  }
-  final def apply(ctx: RuleCtx, patches: Map[RuleName, Patch]): String = {
-    val result = Patch(patches.values.asPatch, ctx, semanticOption)
-    // This overload of apply if purely for convenience
-    // Use fixWithName to iterate over LintMessage
-    // without printing to the console
-    Patch.lintMessages(patches, ctx).foreach(ctx.printLintMessage)
```

Comment:
The old `apply` should have unchanged behavior, otherwise it would be a breaking change.